source/content/guides/composer-convert.md

```diff
@@ -1,156 +1,301 @@
+
+     ```bash{promptUser:user}
+     git checkout master pantheon.yml
+     git add pantheon.yml
+     git commit -m 'Copy my pantheon.yml'
+     ```
 
-`ls` should reveal that the new code repository now has a copy of the `pantheon.upstream.yml`.
+   If you prefer to keep the value for `database` from `pantheon.upstream.yml`, remove it from `pantheon.yml`.
 
 ## Add in the Custom and Contrib Code Needed to Run Your Site
 
 What makes your site code unique is your selection of contributed modules and themes, and any custom modules or themes your development team has created. These customizations need to be replicated in your new project structure.
 
 ### Contributed Code
 
-#### Modules
+#### Modules and Themes
 
-A Composer-managed site should be able to include all custom code via Composer. Begin by reviewing your existing site's code. Check for contributed modules in `/modules`, `/modules/contrib`, `/sites/all/modules`, and `/sites/all/modules/contrib`.
+The goal of this process is to have Composer manage all the site's contrib modules, contrib themes, core upgrades, and libraries (we'll call this _contributed code_). The only things that should be migrated from the existing site are custom code, custom themes, and custom modules that are specific to the existing site.
 
-When reviewing your site, take stock of exactly what versions of modules you depend on. One way to do this is to use a command like the following from within a contributed modules folder (e.g. `/modules`, etc):
+The steps here ensure that any modules and themes from [drupal.org](drupal.org) are in the `composer.json` `require` list.
 
-```bash{outputLines:2-4}
-find . -maxdepth 2 -name '*.info.yml' \
-  -exec basename '{}' '.info.yml' ';' \
-  -exec grep -e '^version' '{}' ';' \
-  -exec echo ';'
-```
+Once Composer is aware of all the contributed code, you'll be able to run `composer upgrade` from within the directory to have Composer upgrade all the contributed code automatically.
 
-This will list each module followed by the version of that module that is installed.
+Begin by reviewing the existing site's code. Check for contributed modules in `/modules`, `/modules/contrib`, `/sites/all/modules`, and `/sites/all/modules/contrib`.
 
-You can add these modules to your new codebase using Composer by running the following for each module in the `$site-composer` directory:
+1. When reviewing the site, take stock of exactly what versions of modules and themes you depend on. One way to do this is to change to run the `pm:projectinfo` Drush command from within a contributed modules folder (e.g. `/modules`, `/themes`, `/themes/contrib`, `/sites/all/themes`, `/sites/all/themes/contrib`, etc.).
 
-```{promptUser:user}
-composer require drupal/MODULE_NAME:^VERSION
-```
+  This command works on Drush 8. If you're using Drush 9, use `pm:list` or refer to [Drush Commands](https://drushcommands.com/drush-9x/pm/pm:projectinfo/):
```

Comment:
Users with Drush 9 are probably on Composer-managed sites, and should be following a different guide. n.b. pm:list shows modules, and pm:projectinfo shows projects. Projects are collections of modules. If the user listed modules, they'd need to figure out the project name from the module name. This is usually the same for projects with one module. Projects with multiple modules often have one with the same name as the project.

Reply:
There is no point in linking to the Drush 9 pm:projectinfo entry on Drush Commands, though; if you run pm:projectinfo you just get an error message telling you to use pm:list or composer show

As of Drush 8.4.8 and Drush 9.7.3, though, you can run `pm:list --fields=*` and see all of the available modules along with the project they're part of. This field isn't included in Drush 10.4.0, but it will be in the next release (hopefully available soon)